src/main/java/org/fungover/haze/Main.java

```diff
@@ -81,21 +108,30 @@ private static void printThreadDebug() {
     }
 
     public static String executeCommand(HazeDatabase hazeDatabase, List<String> inputList, HazeList hazeList) {
-        if (inputList.isEmpty() || inputList.get(0).isEmpty()) {
+        if (inputList.isEmpty() || inputList.get(0).isEmpty())
             return "-ERR no command provided\r\n";
-        }
 
         logger.debug("executeCommand: {} {} ", () -> hazeDatabase, () -> inputList);
-        String command = inputList.get(0).toUpperCase();
 
-        Command commandEnum;
+        Command commandEnum = getCommand(inputList);
+        if (commandEnum == null)
+            return "-ERR unknown command\r\n";
+
+        return commandSwitch(hazeDatabase, inputList, hazeList, commandEnum);
+    }
 
+    private static Command getCommand(List<String> inputList) {
+        String command = inputList.get(0).toUpperCase();
```

Comment:
With java21 there are some new methods on List interface for more readable code. For example getFirst() that can replace get(0).

Reply:
Ok, thank you for input! Will look in to this and issues above